zipkin-cassandra/src/test/scala/com/twitter/zipkin/cassandra/CassandraSpanStoreFactorySpec.scala

```diff
@@ -46,4 +48,32 @@ class CassandraSpanStoreFactorySpec extends FunSuite with Matchers {
       asList(new InetSocketAddress("1.1.1.1", 9143), new InetSocketAddress("2.2.2.2", 9042))
     )
   }
+
+  test("creatingClusterBuilder with SASL authentication null delimited utf8 bytes") {
+    TestFactory.nonExitingMain(Array(
+      "-zipkin.store.cassandra.user", "user",
+      "-zipkin.store.cassandra.password", "pass"
+    ))
+    val SASLhandshake = Array[Byte](0, 'u', 's', 'e', 'r', 0, 'p', 'a', 's', 's')
+    val authProvider = TestFactory.createClusterBuilder()
+                         .getConfiguration()
+                         .getProtocolOptions()
+                         .getAuthProvider()
+                         .newAuthenticator(new InetSocketAddress("localhost", 8080))
+                         .initialResponse()
+    authProvider should be(SASLhandshake)
+  }
+
+  test("creatingClusterBuilder without authentication") {
+    TestFactory.nonExitingMain(Array())
+
+    intercept[AuthenticationException] {
```

Comment:
I like where you're going, but I think it would be clearer to just check it is AuthProvider.NONE?

```
TestFactory.createClusterBuilder()
        .getConfiguration()
        .getProtocolOptions()
        .getAuthProvider() must be (AuthProvider.NONE)
```